dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -120,7 +141,23 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     {
         if (IsPackagedAppScenario())
         {
-            m_channel.PushNotificationReceived(token);
+            if (m_channel)
+            {
+                m_channel.PushNotificationReceived(token);
+            }
+            else
+            {
+                auto lock = m_lock.lock_exclusive();
+
+                wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+                UINT32 appUserModelIdSize{ ARRAYSIZE(appUserModelId) };
+
+                THROW_IF_FAILED(GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId));
```

Comment:
Nit: looks to me to register and unregister a sink, we will first get the appUserModelId. Can we put the logic of getting the appUserModelId into a function, which can then be called in both places?